### PR TITLE
Cards information insert into form inputs

### PR DESCRIPTION
**Pull request by ignaciocervinoGlob**
### Description
Filter information on the cards by the field from the search dialog Link cards to form and charge card information into form inputs, changing the form description to update

#### Tasks:
- [x] Filter information on the cards by the field from the search dialog Link cards to form and charge card information into form inputs
- [x] Change form description when search mode is on, Placeholder for message to switch to update mode or cancel
- [x] Unit Test

#### iPhone 11 Pro
https://user-images.githubusercontent.com/90403672/151048437-3e22d844-8eed-47ee-9792-f161dce3f76d.mp4

#### FormView.setAssetFields
output-onlinepngtools (39)

### CODE REVIEW

File: stockApp/stockApp/Data/Entity/Fields.swift [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Foundation
+
+struct Fields: Codable {
+
+    let values: [String] = ["Barcode", "Asset Name", "User Mail", "Location"]
```
Comment (fn1983) [inline stockApp/stockApp/Data/Entity/Fields.swift:5]: you can have a 

```
enum FieldsType: String, CaseIterable {
        case barcode = "Barcode"
        ...
}

let values: [String] = FieldTypes.allCases.map{$0.rawValue} // or better //  let values: [FieldsType] = FieldTypes.allCases

**Review by fn1983 — APPROVED** [review]
luego reviso cuando tengamos un develop. no deber'iamos hacer rebase de otros branch porque rompe la estructura de arbol de git y no se puede hacer buen code review ya que aparece c'odigo que no esta involucrado en esta tarea

File: stockApp/stockApp/Data/Entity/Fields.swift (hunk 1 above)
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Data/Entity/Fields.swift]: done

File: stockApp/stockApp/Data/Entity/Asset.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+struct Asset: Codable {
+
+    let barcode: String
+    let assetName: String
+    let userMail: String
+    let recordDateTime: String
+    let site: String
+
+    enum CodingKeys: String, CodingKey {
```
[the remark this answers is not in the document]
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Data/Entity/Asset.swift]: done

File: stockApp/Podfile [hunk 3]
```diff
@@ -8,7 +8,7 @@ target 'stockApp' do
   pod 'MaterialComponents/TextControls+FilledTextFields'
   pod 'Firebase/Database'
   pod 'SwiftLint'
-
+	pod 'MaterialComponents/Buttons'
```
[the remark this answers is not in the document]
Comment (ignaciocervinoGlob) [reply stockApp/Podfile:11]: done

[the remark this answers is not in the document]
Comment (ignaciocervinoGlob) [reply stockApp/Podfile:11]: done

**Review by ezequielsetzesglobant — APPROVED** [review]

File: stockApp/stockApp/Presentation/View/AssetView.swift [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import UIKit
+<<<<<<< HEAD
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetView.swift:2]: won't compile!

File: stockApp/stockApp/Presentation/View/AssetView.swift [hunk 5]
```diff
@@ -31,12 +32,47 @@ private struct Constants {
...
+	static let titleLeading: Int = 30
+	static let titleTrailing: Int = 0
+	static let assetsListTop: Int = 30
+	static let assetsListLeading: Int = 10
+	static let assetsListTrailing: Int = -10
+	static let assetsListBottom: Int = -10
+	static let defaultTop: Int = 300
+	static let defaultLeading: Int = 0
+	static let defaultTrailing: Int = 0
+>>>>>>> 3cb6de5 (using MDCFloatingButton())
```
Comment (fn1983) [inline stockApp/stockApp/Presentation/View/AssetView.swift:68]: same! rebase issu!

**Review by fn1983 — CHANGES_REQUESTED** [review]

File: stockApp/stockApp/Presentation/View/AssetView.swift (hunk 5 above)
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/View/AssetView.swift]: done

File: stockApp/stockApp/Presentation/View/AssetView.swift [hunk 6]
```diff
@@ -31,12 +32,47 @@ private struct Constants {
     static let defaultTop: Int = 300
     static let defaultLeading: Int = 0
     static let defaultTrailing: Int = 0
+=======
```
[the remark this answers is not in the document]
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/View/AssetView.swift]: done

**Review by fn1983 — APPROVED** [review]

**Review by Faka-GLB — APPROVED** [review]

**Review by AgustinaB-Globant — APPROVED** [review]

File: stockApp/stockApp/Presentation/View/CameraViewController.swift [hunk 7]
```diff
@@ -9,11 +9,18 @@ class CameraViewController: UIViewController {
...
+    var selectedFilter = Strings.defaultFilter
+    let filters = Strings.filters
+
+    let searchAlertView = SearchAlertView()
 
 	lazy var superView: CameraView = {
-		let view = CameraView(frame: UIScreen.main.bounds)
+        let view = CameraView(frame: UIScreen.main.bounds, isForAddAsset: self.isForAddAsset)
 		view.delegate = self
 		return view
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/CameraViewController.swift:25]: fix alignment

File: stockApp/stockApp/Presentation/View/CameraViewController.swift [hunk 8]
```diff
@@ -104,7 +128,6 @@ class CameraViewController: UIViewController {
 			}
 			let cameraPosition: AVCaptureDevice.Position = strongSelf.isUsingFrontCamera ? .front : .back
 			guard let device = strongSelf.captureDevice(forPosition: cameraPosition) else {
-				HelperFunctions.showAlertMessage(title: Strings.noCameraInput, viewController: self.topMostController() ?? self)
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/CameraViewController.swift:107]: I guess that here are missing code. Error message has to be shown.

File: stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    var onNotAssetsFound: (() -> Void)?
+    var onNoInternetConnection: (() -> Void)?
+	var filter: String
+	var value: String
+
+	init(useCase: GetAssetUseCaseProtocol, filter: String, value: String) {
+        self.useCase = useCase
+		self.filter = filter
+		self.value = value
+    }
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift:18]: fix alignment

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

File: stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift (hunk 9 above)
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/ViewModel/AssetViewModel.swift]: done

File: stockApp/stockApp/Presentation/View/CameraViewController.swift (hunk 8 above)
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/View/CameraViewController.swift]: done

File: stockApp/stockApp/Presentation/View/CameraViewController.swift (hunk 7 above)
Comment (ignaciocervinoGlob) [reply stockApp/stockApp/Presentation/View/CameraViewController.swift]: done

**Review by ezequielvelezglobant — APPROVED** [review]

**Pull request merged**